fix(ngForm): don't block submit when method=dialog

dpogue opened the pull request:
When a form's method is set to "dialog" it does not post to a server but instead closes the nearest parent <dialog> element. An action is not required.

https://html.spec.whatwg.org/multipage/forms.html#submit-dialog

dpogue committed:
fix(ngForm): don't block submit when method=dialog

When a form's method is set to "dialog" it does not post to a server but
instead closes the nearest parent <dialog> element. An action is not required.

dpogue commented:
I signed it!

Narretz commented:
Shocking support for that element: http://caniuse.com/#feat=dialog
But I don't think it'll hurt, as someone who uses `method="dialog"` will know what they're doing. There's still some work to do for this though:
- Add a test for this
- Add a line to the docs that says submit will not be blocked when method is set to dialog.

Narretz commented on src/ng/directive/form.js:
```diff
@@ -485,7 +485,7 @@ var formDirectiveFactory = function(isNgForm) {
 
             // if `action` attr is not present on the form, prevent the default action (submission)
-            if (!('action' in attr)) {
+            if (!('action' in attr) && attr['method'] !== 'dialog') {
```
What happens when the action attr is set and method is set to dialog?

dpogue replied:
According to the spec, if method is dialog then it closes the dialog and stops further processing. The action attribute should be ignored by the browser in that case.

Narretz replied:
What I meant is: if the action attribute is set and the method is set to dialog, your code change would not prevent the submit. Isn't `!('action' in attr) || attr['method'] !== 'dialog' what we want?

Narretz replied:
Ah, I think I misunderstood the purpose. I though it should block submit when the method is dialog, doh. Having the test really helps understanding what a change is about.

dpogue commented:
I've added a test and updated the comments. I'm not sure that blocking submit when method="dialog" with an action is what we want? It should be up to the browser to handle that properly.

Narretz commented:
Thanks for the test. Now that method="dialog" does not block submit anymore, I think we'd still want to have it call $commitViewValue and $setSubmitted on the form, do you agree? Previously, it would set these only if the form wasn't submitted, i.e. sent to the server. Now the form isn't send to the server, so I assume we still want that.
